### ISSUE TITLE
Is 0.4.0 shipped yet?

### ISSUE THREAD

**Issue by jbenet** [ipfs/go-ipfs#2334]
This issue is here so people can subscribe and be notified when they should upgrade. This is important because of the network breaking change.

**Comment by lgierth** [comment]
To answer the question: not yet.

:)

**Comment by RichardLitt** [comment]
:+1: Can't wait!

**Comment by Mithgol** [comment]
https://github.com/notifications → https://github.com/notifications/participating

**Comment by jonnor** [comment]
What is the status? Last release seems to be `0.4.0-rc2` 20 days ago. And seemingly there are features going into master since then?

**Comment by RichardLitt** [comment]
Ping @whyrusleeping

**Comment by whyrusleeping** [comment]
we're just waiting on our new integration test suites to run for a little bit to ensure things are all working smoothly before shipping.

**Comment by whyrusleeping** [comment]
Alright, heres my checklist for shipping 0.4.0:

- [ ] finish blog post (@RichardLitt)
- [ ] write changelog (@whyrusleeping @lgierth @RichardLitt)
- [ ] add final 0.4.0 build to dist.ipfs.io and adjust 'latest' version ( @whyrusleeping or @dignifiedquire )
  - [ ] test that ipfs update shows the 0.4.0 update and can pull/install correctly ( @chriscool @whyrusleeping)
- [ ] ????? ( @jbenet)
- [ ] profit (@ipfs)

**Comment by dignifiedquire** [comment]
- [ ] Prepare new webui release
  - [ ] Merge `feat/files-api` on js-ipfs-api
  - [ ] Update dependencies on webui#master
  - [ ] Ensure geoip works on webui
  - [ ] Make release and upload it
  - [ ] Get at least three people to test it

**Comment by dignifiedquire** [comment]
First webui version to test: http://localhost:5001/ipfs/QmWCJFK5VYXBZqS3LbiRHt8aC3aLSVXU7wNFX9vpckSYun

For testing this run your daemon with `ipfs daemon --unrestricted-api`

**Comment by diasdavid** [comment]
#### Webui testing

- some errors:
![image](https://cloud.githubusercontent.com/assets/1211152/13701518/4fc7694e-e780-11e5-877d-d64182421459.png)
- my browser froze on the logs tab
- shows type as `?` but I'm pretty sure that the second is a dir and the first one is a file (but empty one) 
![image](https://cloud.githubusercontent.com/assets/1211152/13701543/891831ce-e780-11e5-966b-d45c82460281.png)
- what about changing the "GO" button to "RESOLVE"? Kind of feel that it is a good opportunity to 'educate' our users that is what happens when a HASH is searched
![image](https://cloud.githubusercontent.com/assets/1211152/13701564/a6a257a6-e780-11e5-9dbf-10f2922b0a39.png)
- Other than this, looks pretty snappy and fast (I've tested it on a fast machine though), great work!! :+1:

**Comment by dignifiedquire** [comment]
> shows type as ? but I'm pretty sure that the second is a dir and the first one is a file (but empty one)

This hasn't work for some time now (0.3.11, 0.3.10) and this view is going to be removed soon, so I'm not very concerned about fixing this.

> what about changing the "GO" button to "RESOLVE"? Kind of feel that it is a good opportunity to 'educate' our users that is what happens when a HASH is searched image

This release is about making it work with 0.4, I know there are things we want to improve, but that's what the other work we are doing is for.


> my browser froze on the logs tab

Logs are simply too much at the moment, we could remove them for now until we have a better solution, thoughts?

**Comment by dignifiedquire** [comment]
Fixed locales and minification: `QmdmpbMUS1gJ3HppHxFjkUWbV5jpartSi88uUXzBVLBQkU`

**Comment by chpio** [comment]
`pinned files` froze my browser (but that's another issue... again)

**Comment by dignifiedquire** [comment]
> pinned files froze my browser (but that's another issue... again)

Fixing, that is a real issue

**Comment by chpio** [comment]
nah, it's because of the number of entries... pagination would maybe fix this

**Comment by dignifiedquire** [comment]
No more freezing `QmU3o9bvfenhTKhxUakbYrLDnZU7HezAVxPM6Ehjw9Xjqy`

**Comment by chpio** [comment]
lol :D nice fix

**Comment by nginnever** [comment]
Aside from ```Error: Unmapped range(…)``` and a random ```Error: Failed to lookup node(…)``` everything looks 👍

**Comment by whyrusleeping** [comment]
If everyone could test this webui version out asap that would be great. 

Run your ipfs daemon with: `ipfs daemon --unrestricted-api`

and then access http://localhost:5001/ipfs/QmU3o9bvfenhTKhxUakbYrLDnZU7HezAVxPM6Ehjw9Xjqy
in your browser to try it out.

**Comment by noffle** [comment]
Lots of `Failed to load resource: the server responded with a status of 403 (Forbidden)`. I'm running with `--unrestricted-api` on `0.4.0-dev`. UI doesn't appear to be being populated with content.

**Comment by dignifiedquire** [comment]
@noffle does the regular webui work for you? sounds like CORS settings not being correct

**Comment by noffle** [comment]
@dignifiedquire It doesn't. My config:

```
  "API": {
    "HTTPHeaders": {
      "Access-Control-Allow-Credentials": [
        "true"
      ],
      "Access-Control-Allow-Methods": [
        "PUT",
        "GET",
        "POST"
      ],
      "Access-Control-Allow-Origin": [
        "http://localhost:3000"
      ]
    }
  },
```

**Comment by dignifiedquire** [comment]
@noffle add `http://localhost:5001` to the allowed origins restart the daemon, and then navigate to http://localhost:5001/ipfs/QmU3o9bvfenhTKhxUakbYrLDnZU7HezAVxPM6Ehjw9Xjqy

**Comment by noffle** [comment]
@dignifiedquire: thanks, that did the trick.

It's working well for me. I played with all of the affordances and nothing is exploding (violently or otherwise).

**Comment by jbenet** [comment]
@dignifiedquire we should not have to add any CORS stuff to have the webui work by default. it should work out of the box for users. if it doesn't then it needs to be fixed.

**Comment by whyrusleeping** [comment]
@jbenet i think that @noffle had his config changed before testing the new webui leading to the CORS issues he had

**Comment by dignifiedquire** [comment]
@jbenet yes it works out of the box, but @noffle had a non default config, so it didn't work.

**Comment by jbenet** [comment]
@dignifiedquire ah ok makes sense. we should probably warn users that setting an origin like that will mess with the default expectations (or maybe we should just make the default config show the expectations)

**Comment by noffle** [comment]
@dignifiedquire Are you sure? I did a fresh `ipfs init` and see this `"API"` entry in the config:

```
  "API": {
    "HTTPHeaders": null
  },
```

**Comment by dignifiedquire** [comment]
@noffle works fine for me on `master` on a fresh init. Even though headers is empty, the default is set internally I think.

**Comment by jbenet** [comment]
# Yes. 0.4.0 shipped. 

http://blog.ipfs.io/14-ipfs-0-4-0-released/

![](https://ipfs.io/ipfs/QmXG2LtKSWaRBw2yEbN1H7aZGC5f57ZmJFJmLSkFAL5LWo/0.4.0-shipped.jpg)

**Comment by jbenet** [comment]
# Thanks to everyone who contributed to this huge release
